### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by hohy**
Updated dependencies and Node.js version used in build pipeline.
Code reformatted using updated `prettier`.

## How Has This Been Tested?

Unit and integration tests.

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by hohy** [comment]
@slavek-kucera Ready for review

**Comment by slavek-kucera** [comment]
There is still a bunch of warnings of GitHub Actions depending on node16.

File: clients/cobol-lsp-vscode-extension/package.json [hunk 1]
```diff
@@ -584,7 +584,7 @@
   "scripts": {
     "vscode:prepublish": "npm run -S build -- --minify && npm run -S build:web -- --production -- --minify",
     "postinstall": "node -e \"if (process.env.NODE_ENV === 'production'){process.exit(1)} \" && patch-package || echo skip patching",
-    "build": "esbuild src/extension.ts --bundle --sourcemap --platform=node --target=node16.17.1 --outdir=dist --external:vscode --format=cjs",
+    "build": "esbuild src/extension.ts --bundle --sourcemap --platform=node --target=node20 --outdir=dist --external:vscode --format=cjs",
```
Comment (slavek-kucera) [inline clients/cobol-lsp-vscode-extension/package.json]: is the `node20` target compatible with the oldest supported VSCode version from `vscode-languageclient` which is `1.82.0`?
It looks like VSCode update to node20 in `1.90.0`.

Comment (hohy) [reply clients/cobol-lsp-vscode-extension/package.json]: I see we have in package.json
```
  "engines": {
    "vscode": "^1.71.0"
  },
  ```
So we should theoretically support vscode 1.71.
Is there some agreement or rule on how old vscode versions we have to support? 
I see the version 1.90 was released in May 2024, so more than 6 months ago. Is it enough or we need to suport older versions as well? 
The version 1.82 is from August 2023.

Comment (slavek-kucera) [reply clients/cobol-lsp-vscode-extension/package.json]: Vít say 6 versions. Are there any issues when targeting node18?

Comment (hohy) [reply clients/cobol-lsp-vscode-extension/package.json]: No issues with node18, I'll set it as a target and set minimal vscode version to 1.82 👍

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**